app/views/LocationController.js

```diff
@@ -0,0 +1,114 @@
+import BackgroundGeolocation from '@mauron85/react-native-background-geolocation';
+import React, { Component } from 'react';
+import {
+  AppState,
+  BackHandler,
+  Image,
+  StatusBar,
+  TouchableOpacity,
+  View,
+} from 'react-native';
+import { SvgXml } from 'react-native-svg';
+
+import settingsIcon from './../assets/svgs/settingsIcon';
+import { isPlatformAndroid } from './../Util';
+import Colors from '../constants/colors';
+import LocationServices from '../services/LocationService';
+import DefaultPage from './tracking/DefaultPage';
+import ExposurePage from './tracking/ExposurePage';
+import OffPage from './tracking/OffPage';
+import styles from './tracking/style';
+import UnknownPage from './tracking/UnknownPage';
+
+export default class LocationController extends Component {
+  constructor(props) {
+    super(props);
+
+    if (isPlatformAndroid()) {
+      StatusBar.setBackgroundColor(Colors.TRANSPARENT);
+      StatusBar.setBarStyle('light-content');
+      StatusBar.setTranslucent(true);
+    }
+
+    this.state = {
+      location: {
+        canTrack: true,
+        reason: '',
+        crossedPaths: false,
+      },
+    };
+  }
+
+  componentDidMount() {
+    this.updateStateInfo();
+    // refresh state if user backgrounds app
+    AppState.addEventListener('change', () => {
+      this.updateStateInfo();
+    });
+    // refresh state if settings change
+    this.unsubscribe = this.props.navigation.addListener('focus', () => {
+      this.updateStateInfo();
+    });
+    // handle back press
+    BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
+  }
+
+  componentWillUnmount() {
+    console.log('unmounting component');
+    AppState.removeEventListener('change', this.handleAppStateChange);
+    BackHandler.removeEventListener('hardwareBackPress', this.handleBackPress);
+    this.unsubscribe();
+  }
+
+  handleBackPress = () => {
+    BackHandler.exitApp(); // works best when the goBack is async
+    return true;
+  };
+
+  getSettings() {
+    return (
+      <TouchableOpacity
+        style={styles.settingsContainer}
+        onPress={() => {
+          this.props.navigation.navigate('SettingsScreen');
+        }}>
+        {/* Is there is a reason there's this imageless image tag here? Can we delete it? */}
+        <Image resizeMode={'contain'} />
```

Comment:
delete it. I think it was an attempt tofix a bug in ios 12. which we fixed in another way.